src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -115,6 +115,21 @@ public function merge($items)
     }
 
     /**
+     * Run a map over each of the items.
+     *
+     * @param  callable  $callback
+     * @return static
+     */
+    public function map(callable $callback)
+    {
+        $result = parent::map($callback);
+
+        return $result->contains(function ($_, $item) {
+            return ! ($item instanceof Model);
```

Comment:
style habit of mine, I'e never thought the operator precedence of instanceof was very intuitive. I won't put up a fuss if no one else likes this though :)